clientgui/BOINCGUIApp.cpp

```diff
@@ -891,23 +898,48 @@ void CBOINCGUIApp::DetectDataDirectory() {
 
 
 void CBOINCGUIApp::InitSupportedLanguages() {
-    wxInt32               iIndex = 0;
-    const wxLanguageInfo* liLanguage = NULL;
-
-    // Prepare the array
-    m_astrLanguages.Insert(wxEmptyString, 0, wxLANGUAGE_USER_DEFINED+1);
-
-    // These are just special tags so deal with them in a special way
-    m_astrLanguages[wxLANGUAGE_DEFAULT]                    = _("(Automatic Detection)");
-    m_astrLanguages[wxLANGUAGE_UNKNOWN]                    = _("(Unknown)");
-    m_astrLanguages[wxLANGUAGE_USER_DEFINED]               = _("(User Defined)");
+    // Find available translations
+    std::vector<const wxLanguageInfo*> langs;
+    const wxTranslations* pTranslations = wxTranslations::Get();
+    if (pTranslations) {
+        wxArrayString langCodes = pTranslations->GetAvailableTranslations(wxT("BOINC-Manager"));
+        for (const auto& langCode : langCodes) {
+            const wxLanguageInfo* pLI = wxLocale::FindLanguageInfo(langCode);
+            if (pLI) {
+                langs.push_back(pLI);
+            }
+        }
+    }
 
-    for (iIndex = 0; iIndex <= wxLANGUAGE_USER_DEFINED; iIndex++) {
-        liLanguage = wxLocale::GetLanguageInfo(iIndex);
-        if (liLanguage) {
-            m_astrLanguages[iIndex] = liLanguage->Description;
+    // Synthesize labels to be used in the options dialog
+    m_astrLanguages.reserve(langs.size() + 1);  // +1 for the entry for "English"
+    bool has_translation_en = false;
+    for (const auto& pLI : langs) {
+        wxString Label =
+#if wxCHECK_VERSION(3,1,6)
```

Comment:
This is mean 3.1.6 or higher? On all other os we already at 3.2.1.1

Reply:
Yes: the native language description was added in wxWidgets 3.1.6

Reply:
We have now wxWidgets 3.1.5 for OSX. It's highly possible that in the future release we will update it to 3.1.6 or later.
So please keep this PR in draft, so we can come back to it when time comes.